### DISCUSSION TITLE
Can't get SemanticDB to work for amm scripts

### DISCUSSION THREAD

**Discussion by CherryWorm** [scalameta/metals#4126] [Q&A]
In a clean directory, creating an empty `test.sc` file and importing it with the latest stable version of metals in vs code (v1.18.0, metals version 0.11.7), Metals Doctor will show that the Scala version of the target is 2.13.7 (even though I explicitly set the fallback Scala version to 3.1.1) and SemanticDB is not available.

Anyone have a pointer on how I can get Ammonite scripts to work properly?

**Comment by tgodzik** [comment]
Ammonite will use a default version of Scala unless it's specified in the file itself. You should be able to add:
```
// scala 3.1.1
```
but unfortunately Ammonite runner is still not supported for Scala 3, it's stuck on the PR https://github.com/alexarchambault/ammonite-runner/pull/127

If you want to work with Scala 3 I recommend using ScalaCLI https://scala-cli.virtuslab.org/ or staying with ammonite for Scala 2

As for semanticdb, it should get compiled properly for the file, but if you send us a sample we can check it out.

**Reply by CherryWorm** [reply]
Ah right, setting the scala version that way unfortunately fails because it seems like `ammonite-repl-api` hasn't been published for scala 3 though.

What kind of sample do you need? The directory contains nothing but an empty script file

**Reply by tgodzik** [reply]
So semanticdb should not be an issue really unless references/renames doesn't work?
